### PR TITLE
Use cabal-v2 on CI

### PR DESCRIPTION
**Pull request by sol**

### CODE REVIEW

**Comment by andreasabel** [comment]
Looks like the GHC-8 tests fail because GHCi chatters about `Loading package environment...`.
The GHC-7 tests have the problem `Not in scope: polyQuickCheck`.

**Comment by sol** [comment]
> Looks like the GHC-8 tests fail because GHCi chatters about `Loading package environment...`.

That's a bug in those versions of GHC.  They don't obey `-v0`.  I pushed 
4a72f3d as a fix.

File: package.yaml [hunk 1]
```diff
@@ -18,19 +15,6 @@ maintainer:       quasicomputational@gmail.com, Andreas Abel
 
 github: sol/doctest
 
-tested-with:
```
Comment (andreasabel) [inline package.yaml:21]: You don't like this field?

Comment (sol) [reply package.yaml:21]: AFAIK, this field is only used by `haskell-ci`.  So unless I'm using `haskell-ci` I can't justify burning any cycles on keeping this up to date.

**Pull request merged**